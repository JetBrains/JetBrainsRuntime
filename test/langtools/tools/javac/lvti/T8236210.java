/*
 * @test /nodynamiccopyright/
 * @bug 8236210
 * @summary Annotations on var
 * @compile/fail/ref=T8236210.out -XDrawDiagnostics T8236210.java
 */

import static java.lang.annotation.ElementType.LOCAL_VARIABLE;
import static java.lang.annotation.ElementType.TYPE_USE;

import java.lang.annotation.Target;

@Target({TYPE_USE, LOCAL_VARIABLE})
@interface Both {}

@Target(LOCAL_VARIABLE)
@interface Decl {}

@Target(TYPE_USE)
@interface Type {}

class T8236210 {
    {
        @Both var a = 1;
        @Decl var b = 2;
        @Type var c = 3;
    }
}
