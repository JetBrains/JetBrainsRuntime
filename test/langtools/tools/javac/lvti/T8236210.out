T8236210.java:26:9: compiler.err.annotation.type.not.applicable
1 error
